Sources/Basic/FileSystem.swift

```diff
@@ -279,67 +296,13 @@ private class LocalFileSystem: FileSystem {
     }
 
     func getDirectoryContents(_ path: AbsolutePath) throws -> [String] {
-        guard let dir = SPMLibc.opendir(path.pathString) else {
-            throw FileSystemError(errno: errno)
-        }
-        defer { _ = SPMLibc.closedir(dir) }
-
-        var result: [String] = []
-        var entry = dirent()
-
-        while true {
-            var entryPtr: UnsafeMutablePointer<dirent>? = nil
-
-            let readdir_rErrno = readdir_r(dir, &entry, &entryPtr)
-            if  readdir_rErrno != 0 {
-                throw FileSystemError(errno: readdir_rErrno)
-            }
-
-            // If the entry pointer is null, we reached the end of the directory.
-            if entryPtr == nil {
-                break
-            }
-
-            // Otherwise, the entry pointer should point at the storage we provided.
-            assert(entryPtr == &entry)
-
-            // Add the entry to the result.
-            guard let name = entry.name else {
-                throw FileSystemError.invalidEncoding
-            }
-
-            // Ignore the pseudo-entries.
-            if name == "." || name == ".." {
-                continue
-            }
-
-            result.append(name)
-        }
-
-        return result
+        return try FileManager.default.contentsOfDirectory(atPath: path.pathString)
     }
 
     func createDirectory(_ path: AbsolutePath, recursive: Bool) throws {
-        // Try to create the directory.
-        let result = mkdir(path.pathString, SPMLibc.S_IRWXU | SPMLibc.S_IRWXG)
-
-        // If it succeeded, we are done.
-        if result == 0 { return }
-
-        // If the failure was because the directory exists, everything is ok.
-        if errno == EEXIST && isDirectory(path) { return }
-
-        // If it failed due to ENOENT (e.g., a missing parent), and we are
-        // recursive, then attempt to create the parent and retry.
-        if errno == ENOENT && recursive &&
-           path != path.parentDirectory /* FIXME: Need Path.isRoot */ {
-            // Attempt to create the parent.
-            try createDirectory(path.parentDirectory, recursive: true)
-
-            // Re-attempt creation, non-recursively.
-            try createDirectory(path, recursive: false)
-        } else {
-            // Otherwise, we failed due to some other error. Report it.
+        do {
+            try FileManager.default.createDirectory(atPath: path.pathString, withIntermediateDirectories: recursive)
+        } catch {
             throw FileSystemError(errno: errno)
         }
```

Comment:
You should be able to catch an `_NSErrorWithErrno` and extract the `errno` from that as it may be overwritten in between.

Reply:
Yeah, however, it seems that there is no real reason to keep the custom error wrappers around, so, Im going to change towards letting the `NSError` propagate through which makes things simpler and less likely to introduce errors like that.